move-mutation-test/src/cli.rs

```diff
@@ -0,0 +1,204 @@
+/// from the mutation-test options which does not allow setting it.
+#[must_use]
+pub fn check_mutator_output_path(options: &move_mutator::cli::CLIOptions) -> Option<PathBuf> {
+    options
+        .configuration_file
+        .as_ref()
+        .and_then(|conf| move_mutator::configuration::Configuration::from_file(conf).ok())
+        .and_then(|c| c.project.out_mutant_dir)
+}
+
+/// The configuration options for running the tests.
+// Info: this set struct is based on TestPackage in `aptos-core/crates/aptos/src/move_tool/mod.rs`.
+#[derive(Parser, Debug)]
+pub struct TestBuildConfig {
+    /// Options for compiling a move package dir.
+    // We might move some options out and have our own option struct here - not all options are
+    // needed for mutation testing.
+    #[clap(flatten)]
+    pub move_pkg: MovePackageDir,
+
+    /// Dump storage state on failure.
+    #[clap(long = "dump")]
+    pub dump_state: bool,
+
+    /// A filter string to determine which unit tests to run.
+    #[clap(long, short)]
+    pub filter: Option<String>,
+
+    /// A boolean value to skip warnings.
+    #[clap(long)]
+    pub ignore_compile_warnings: bool,
+    // TODO: Unused in aptos-core:
+    ///// The maximum number of instructions that can be executed by a test
+    /////
+    ///// If set, the number of instructions executed by one test will be bounded
+    //#[clap(long = "instructions", default_value_t = 100000)]
+    //pub instruction_execution_bound: u64,
+
+    // TODO: There is no sense in enabling coverage - we'll have another option in the future
+    // 'use-coverage-data' or something like that - that is going to be passed to the mutator
+    // tool
+    ///// Collect coverage information for later use with the various `aptos move coverage` subcommands
+    //#[clap(long = "coverage")]
+    //pub compute_coverage: bool,
+}
+
+impl TestBuildConfig {
+    pub fn compiler_config(&self) -> CompilerConfig {
+        let known_attributes = extended_checks::get_all_attribute_names().clone();
+        CompilerConfig {
+            known_attributes,
+            skip_attribute_checks: self.move_pkg.skip_attribute_checks,
+            bytecode_version: get_bytecode_version(
+                self.move_pkg.bytecode_version,
+                self.move_pkg.language_version,
+            ),
+            compiler_version: self.move_pkg.compiler_version,
+            language_version: self.move_pkg.language_version,
+            experiments: experiments_from_opt_level(&self.move_pkg.optimize),
+        }
+    }
+}
+
+/// Get bytecode version.
+fn get_bytecode_version(
+    bytecode_version_in: Option<u32>,
+    language_version: Option<LanguageVersion>,
+) -> Option<u32> {
+    bytecode_version_in.or_else(|| language_version.map(|lv| lv.infer_bytecode_version(None)))
+}
+
+/// Get a list of stringified [`Experiment`] from the optimization level.
+fn experiments_from_opt_level(optlevel: &Option<OptimizationLevel>) -> Vec<String> {
```

Comment:
Could this be used from the crate instead of implementing here?

Reply:
We could, but we need to make it public in `aptos-core`. I can do that within the same PR (related to the other comment below):

> I left a reference to this aptos-core/crates/aptos/src/move_tool/mod.rs - this function is based upon TestPackage::execute method mostly/